test/Snail/IOSpec.hs

```diff
@@ -0,0 +1,27 @@
+module Snail.IOSpec where
+
+import Data.Either
+import Snail.IO
+import Test.Hspec
+
+spec :: Spec
+spec = do
+  describe "successfully lexes some basic snail files" $ do
+    it "lex a basic snail file" $ do
+      eResults <- readSnailFile "examples/basic.snail"
+      eResults `shouldSatisfy` isRight
+
+    -- Not concerned with this at the lexing stage, we can
+    -- error at the validation phase or just treat this
+    -- as '()'
```

Comment:
This comment makes no sense anymore. I switched to `sepEndBy1` to avoid passing an empty file